### PR TITLE
Use prod buckets

### PR DESCRIPTION
**Pull request by BobyMCbobs**
- adds useful documentation for managing the buckets
- ensures the use of the registry.k8s.io AWS account
- updates the default bucket name
- removes the iam role

Co-Authored-By: jaypipes <jaypipes@gmail.com>

### CODE REVIEW

**Review by jaypipes — APPROVED** [review]
yay2

Production buckets are now all created (see pic above). :rocket:

File: infra/aws/terraform/registry.k8s.io/s3/bucket.tf [hunk 1]
```diff
@@ -60,29 +52,15 @@ resource "aws_s3_bucket_policy" "registry-k8s-io-public-read" {
   })
 }
 
-resource "aws_iam_user_policy" "registry-k8s-io-rw" {
-  name = "${aws_s3_bucket.registry-k8s-io.bucket}-access"
-  user = var.iam_user_name
```
Comment (jaypipes) [inline infra/aws/terraform/registry.k8s.io/s3/bucket.tf:65]: note to reviewers: this IAM User policy should have been removed in an earlier commit (see PR #3648) but we only noticed it just now.

**Comment by Riaankl** [comment]
Thanks for the review.
Do we need more eyes or can we `/unhold` @ameukam @dims

**Comment by ameukam** [comment]
> Thanks for the review. Do we need more eyes or can we `/unhold` @ameukam @dims

Feel free to remove the `hold` when you want.

**Comment by ameukam** [comment]
According to https://github.com/kubernetes/k8s.io/issues/3595#issuecomment-1089564784 and https://github.com/kubernetes-sigs/oci-proxy/issues/38#issuecomment-1085369923, `eu-west-1` is missing. Completely missed it.

**Comment by jaypipes** [comment]
> According to [#3595 (comment)](https://github.com/kubernetes/k8s.io/issues/3595#issuecomment-1089564784) and [kubernetes-sigs/oci-proxy#38 (comment)](https://github.com/kubernetes-sigs/oci-proxy/issues/38#issuecomment-1085369923), `eu-west-1` is missing. Completely missed it.

@ameukam https://github.com/kubernetes/k8s.io/pull/3699

**Pull request merged**